### DISCUSSION TITLE
How to detect a pull request from a fork

### DISCUSSION THREAD

**Discussion by jamime** [community/community#25217] [Actions]
<p>I have a project that uses cypress dashboard. To access this we use 20 runners (our max concurrency) and a secret token.</p>
<p>This uses</p>
<pre><code>fail-fast: false
</code></pre>
<p>otherwise when one runner fails they will terminate without updating the cypress dashboard.<br>
<a href="https://github.com/cypress-io/github-action/issues/48" rel="nofollow noopener">https://github.com/cypress-io/github-action/issues/48</a></p>
<p>When we recieve a PR from a fork it takes up all 20 of our available runners, but only uses 1 (because it does not have the secret token). This takes ~3 hours.</p>
<p>Is it possible to detect if the PR came from a fork so I can adjust the matrix size?</p>

**Comment by jamime** [comment, marked as answer]
<p>I found the following works (ensure you replace org/repo).</p>
<p>Fork:</p>
<pre><code>if: ${{ github.event.pull_request.head.repo.full_name != 'org/repo' }}
</code></pre>
<p>Not a fork:</p>
<pre><code>if: ${{ github.event.pull_request.head.repo.full_name == 'org/repo' }}
</code></pre>

**Comment by tunetheweb** [comment]
<p>This also seems to work which is more generic, than actually naming the repo:</p>
<pre><code>if: github.event.pull_request.head.repo.full_name == github.repository
</code></pre>

**Comment by felipecrs** [comment]
<p>In case it’s useful for someone, I wanted to filter <code>pull_requests</code> which does not have write permissions, which are:</p>
<ul>
<li>PRs coming from forks</li>
<li>PRs coming from dependabot</li>
</ul>
<p>I was able to do so by using:</p>
<pre><code class="lang-yaml">if: github.event.pull_request.head.repo.full_name == github.repository &amp;&amp; github.actor != 'dependabot[bot]'
</code></pre>

**Reply by eiffel-fl** [reply]
Already found the dependabot part, but I was looking for the fork part! Thank you for it!

**Comment by islami00** [comment]
<p>Hi, you can now check if fork directly using the boolean provided in the payload:<br>
Not fork:<br>
<code>if: "! github.event.pull_request.head.repo.fork " </code><br>
Fork:<br>
<code>if: " github.event.pull_request.head.repo.fork "</code></p>

**Reply by gluxon** [reply]
I heads up to others trying this: `github.event.pull_request.head.repo.fork` describes the repo making the PR, not the PR itself. This value will be `true` for any PRs from a forked repo, even if the PR is targeting the fork itself.

If you're looking for whether a pull request has a head branch from a different repo, an earlier answer is probably better.

- https://github.com/orgs/community/discussions/25217#discussioncomment-3246902
